src/azul/plugins/repository/canned/__init__.py

```diff
@@ -161,20 +274,20 @@ def fetch_bundle(self, bundle_fqid: CannedBundleFQID) -> CannedBundle:
     def portal_db(self) -> Sequence[JSON]:
         return []
 
-    def _construct_file_url(self, source_url: str, file_name: str) -> str:
+    def _construct_file_url(self, spec: CannedSourceSpec, file_name: str) -> str:
         """
-        >>> source_url = 'https://github.com/USER/REPO/tree/REF/tests'
-        >>> Plugin._construct_file_url(Plugin, source_url, 'foo.zip')
-        'https://github.com/USER/REPO/raw/REF/tests/foo.zip'
+        >>> spec = CannedSourceSpec.parse('https://github.com/user/repo.git::f0f0f0f0::dir/subdir::/0')
+        >>> self = Plugin.create_empty()
+        >>> Plugin._construct_file_url(self, spec, 'foo.zip')
+        'https://github.com/user/repo/raw/f0f0f0f0/dir/subdir/foo.zip'
 
-        >>> Plugin._construct_file_url(Plugin, source_url, '')
+        >>> Plugin._construct_file_url(self, spec, '')
         Traceback (most recent call last):
         ...
         azul.RequirementError: file_name cannot be empty
         """
-        url = furl(source_url)
-        require(url.path.segments[2] == 'tree', source_url)
-        url.path.segments[2] = 'raw'
+        url = furl(spec.url[:-4])
```

Comment:
This is the intended behavior, correct?

```suggestion
        url = furl(spec.url.removesuffix('.git'))
```